nextcord/application_command.py

```diff
@@ -1240,76 +1262,180 @@ def __init__(
+                        found_choices.append(lit)
+
+                else:
+                    raise ValueError(
+                        f"{self.error_name} Invalid type for choices: {type(lit)}: {lit}"
+                    )
+
+            if found_type is MISSING:
+                # ??? Why are they using Literal???
+                found_type = ApplicationCommandOptionType.string
+
+            annotation_type = found_type
+            annotation_choices = found_choices
+        elif typehint_origin in (Union, Optional, None):
+            # If the typehint base is Union, Optional, or not any grouping...
+            found_type = MISSING
+            found_channel_types = []
+
+            if typehint_origin is None:
+                unpacked_annotations: List[type] = [
+                    parameter.annotation,
+                ]
+                literals: List[Annotated[OptionConverter, object]] = []
+            else:
+                unpacked_annotations, literals = unpack_annotation(
+                    parameter.annotation, list(self.option_types.keys())
+                )
+            # Make sure that all literals are only OptionConverters and nothing else.
+            for lit in literals:
+                if not isinstance(lit, OptionConverter) or lit is not None:
+                    raise ValueError(
+                        f"{self.error_name} You cannot use non-OptionConverter literals when the base annotation is "
+                        f"not Literal."
+                    )
+
+            # Pyright gets upset at appending these lists together, but not upset when .extend is used?
+            # grouped_annotations: List[Union[type, Annotated[object, OptionConverter]]] = unpacked_annotations + \
+            #                                                                              literals
+            grouped_annotations: List[
+                Union[type, Annotated[Optional[OptionConverter], object], Type[None]]
+            ] = []
+            grouped_annotations.extend(unpacked_annotations)
+            grouped_annotations.extend(literals)
+            # The only literals in this should be OptionConverters. Anything else should have triggered the ValueError.
+            for anno in grouped_annotations:
+                if isinstance(anno, object) and isinstance(anno, OptionConverter):
+                    # If the annotation is instantiated, add it to the converters and set the anno to the type it has.
+                    annotation_converters.append(anno)
+                    anno = anno.type
+                elif isinstance(anno, type) and issubclass(anno, OptionConverter):
+                    # If the annotation is NOT instantiated, instantiate it and do the above.
+                    made_converter = anno()
+                    annotation_converters.append(made_converter)
+                    anno = made_converter.type
+
+                if anno is None or anno == type(None):  # type: ignore
```

Comment:
Using the `is` operator worked fine for me
```python
>>> anno = type(None)
>>> anno is type(None)
True
```

Reply:
I'll need to run more tests, but on Python 3.10 for me it didn't work fine.

Reply:
```ps
oliver in ~ λ py
Python 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> type(None) is type(None)
True
>>> h = type(None)
>>> h is type(None)
True
>>> 
```

Reply:
Switching the `==` to `is` did work, I swear I was having a huge amount of issues with it earlier.